lib/rubocop/cop/style/yoda_condition.rb

```diff
@@ -67,9 +67,14 @@ class YodaCondition < Cop
 
         NONCOMMUTATIVE_OPERATORS = %i[===].freeze
 
+        def_node_matcher :file_constant_equal_doller_zero?, <<~PATTERN
+          (send #source_file_path_constant? :== (gvar #doller_zero?))
```

Comment:
Shouldn't we cover some of the other comparison operators as well? `<` and friends probably don't make sense, but I could imagine seeing `__FILE__ != $0` in the wild.

And what about the English name of `$0`, that is `$PROGRAM_NAME`?

Reply:
Sure! I change to accept `!=` and `$PROGRAM_NAME`.